test-data/unit/check-dataclass-transform.test

```diff
@@ -44,3 +44,20 @@ Person('Jonh', 21, None)  # E: Too many arguments for "Person"
 
 [typing fixtures/typing-full.pyi]
 [builtins fixtures/dataclasses.pyi]
+
+[case testDataclassTransformParametersAreApplied]
+# flags: --python-version 3.7
+from typing import dataclass_transform, Callable, Type
+
+@dataclass_transform()
+def my_dataclass(*, eq: bool, order: bool) -> Callable[[Type], Type]:
+    def transform(cls: Type) -> Type:
+        return cls
+    return transform
+
+@my_dataclass(eq=False, order=True)
+class Application:  # E: eq must be True if order is True
+   ...
```

Comment:
Here are some ideas for additional tests:
* Test that `__init__` gets generated correctly if there are attributes.
* Test that e.g. `@my_dataclass(eq="x")` generates an error.

Reply:
> Test that e.g. `@my_dataclass(eq="x")` generates an error.

The existing dataclass code enforces that parameters are boolean literals this but apparently doesn't exercise that in its tests; I will at least add some to the new test file, though.

For `dataclasses.dataclass` and `typing.dataclass_transform` we can rely on the regular type-checking pass alongside the typeshed stubs, but I guess we can't do that for the user transforms; should we just rely on these checks for that? I guess it would be hard to enforce it with typeshed since (AFAIK) there's not an easy way to define a `Callable` with optional kwargs that should have a specific type if present.